daemon/images/image.go

```diff
@@ -46,6 +47,11 @@ type manifest struct {
 	Config specs.Descriptor `json:"config"`
 }
 
+func (i *ImageService) PrepareSnapshot(ctx context.Context, id string, image string, platform *v1.Platform) error {
+	// Only makes sense when conatinerd image store is used
+	panic("not implemented")
```

Comment:
I realize this should only be invoked if we've made a programming error in the engine, but `panic` seems like a pretty abrasive way to handle that -- wouldn't it make sense for this to return some basic error instead? :sweat_smile:
(Perhaps even collecting a backtrace in the error?  I guess I can see why you chose `panic` instead, it just feels very heavy since it'll take down the whole daemon :see_no_evil:)

Reply:
You are right... My thinking was "ok this must not happen, let's fail spectacularly if it does", but maybe that's too harsh :)

Reply:
:shrug: let's start here, not backport this to 23.0, and see what happens :joy:

Hopefully if we've messed up before this codepath is removed, someone will notice before it hits a release. :smile: